solver/src/driver.rs

```diff
@@ -350,10 +350,13 @@ impl Driver {
         }
 
         let rated_settlements = self.rate_settlements(settlements, &estimated_prices).await;
+        // TODO: 1e18 should be 10**(native_token_decimals)
+        let gas_price_normalized: BigRational = estimated_prices.get(&self.native_token).unwrap()
+            / BigRational::from_integer(1_000_000_000_000_000_000_u128.into());
 
         if let Some(mut settlement) = rated_settlements.into_iter().max_by(|a, b| {
-            a.objective_value(gas_price)
-                .cmp(&b.objective_value(gas_price))
+            a.objective_value(&gas_price_normalized)
```

Comment:
Maybe we should change the way we compute the objective value from doing it in two places (once on RatedSettlement assuming surplus has already been cached) and once on Settlement directly so that we have one method where we can unit test our expectations given internal prices external prices and gas price?

Reply:
I tried to address this in the commit below, but not sure I did (also added a unit test for that function, hopefully with real world values).